.github/workflows/gradle.yml

```
@@ -23,45 +33,42 @@ jobs:
       run: git submodule update --init --single-branch
     - name: Set up Java
       uses: actions/setup-java@dded0888837ed1f317902acf8a20df0ad188d165 # v5.0.0
-      # By putting JVMs in reverse order, the last one is the one that
-      # will go in the path.
       with:
+        # By putting JVMs in reverse order, the last one is the one that
+        # will go in the path.
         java-version: |
-          11
-          17
+          ${{ matrix.java-version }}
           21
         distribution: 'adopt'
-    - name: Build and test Java 21
-      run: >-
-        ./gradlew check
-    - name: Upload results Java 21
-      uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
-      if: ${{ always() }}
-      with:
-        name: reports-java-21
-        path: '*/build/reports'
-    - name: Build and test Java 17
+    - name: Build and test ${{ matrix.name }}
       env:
-        RHINO_TEST_JAVA_VERSION: "17"
+        RHINO_TEST_JAVA_VERSION: ${{ matrix.java-version }}
+      timeout-minutes: 30
       run: >-
-        ./gradlew check
-    - name: Upload results Java 17
+        ./gradlew check ${{ matrix.multithread && '-Drhino.useThreadSafeObjectsByDefault=true' || '' }}
+    - name: Upload results ${{ matrix.name }}
       uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
       if: ${{ always() }}
       with:
-        name: reports-java-17
+        name: reports-${{ format('java-{0}', matrix.java-version) }}${{ matrix.multithread && '-multithread' || '' }}
         path: '*/build/reports'
-    - name: Build and test Java 11
-      env:
-        RHINO_TEST_JAVA_VERSION: "11"
-      run: >-
-        ./gradlew check
-    - name: Upload results Java 11
-      uses: actions/upload-artifact@ea165f8d65b6e75b540449e92b4886f43607fa02 # v4.6.2
-      if: ${{ always() }}
+
+  test262:
+    runs-on: ubuntu-latest
+    name: Test262 Suite
```

I think this step should be called
```suggestion
    name: Check if test262.properties is up-to-date
```
See: https://github.com/mozilla/rhino/pull/2052/files